### ISSUE TITLE
<machine_name> does not crash when there is another machine called <machine_name>-<something>

### ISSUE THREAD

**Issue by Oshawk** [netkit-jh/netkit-jh-build#77]
**Describe the bug**
When there is a machine called `<machine_name>` and another machine called `<machine_name>-<something>`, running `lcrash` only crashes the latter machine. Running it again crashes the former.

**To Reproduce**
```
mkdir test
cd test
touch {a,a-a}.startup lab.{conf,dep}
mkdir a a-a
lstart
lcrash
```

**Expected behavior**
Both machines crash.

**Operating System**
- Host OS: Windows 10
- Guest OS: Ubuntu 20.04

**Additional context**
This seems to be an issue with the `-` character. Crashing the machines `a` and `aa` works fine.

**Comment by EdOverflow** [comment]
Nice catch, @Oshawk! @AdamBromiley and I played around with it and discovered the root cause. It turns out any non-word character causes trouble. The root cause can be found in the `getVMinfoByName()` function in `core/bin/script_utils`.

https://github.com/netkit-jh/netkit-jh-build/blob/d6951ce92cf9aaf0064f4adfac72cc9348d0f34a/core/bin/script_utils#L389-L397

`grep -w "^$1 .*umid=$2"` will of course match both `a` and `a-a` when `$2` equates to `a` due to the `-w` flag which skips the `-` character.

```sh
$ echo "a-a" | grep -w "a"
a-a
$ echo "a" | grep -w "a"
a
```

This combined with `head -n 1` results in the process with the lowest PID being crashed only. This explains why sometimes netkit is able to crash both correctly since the behaviour you described depends on which order the VMs were spawned in.

@AdamBromiley will submit a PR shortly.

**Comment by TechSupportJosh** [comment]
Fixed by #78
